rococo/repositories/mongodb/mongodb_repository.py

```diff
@@ -0,0 +1,145 @@
+            queue_name: str,
+            user_id: UUID = None
+    ):
+        super().__init__(db_adapter, model, message_adapter, queue_name, user_id=user_id)
+        self.db = db_adapter.db
+
+
+    def _process_data_before_save(self, instance: VersionedModel):
+        """Method to convert VersionedModel instance to a data dictionary that can be inserted in MongoDB"""
+        super()._process_data_before_save(instance)
+        data = instance.as_dict(convert_datetime_to_iso_string=False, convert_uuids=True)
+        for field in fields(instance):
+            if data.get(field.name) is None:
+                continue
+
+            field_value = data[field.name]
+
+            if field.metadata.get('field_type') in ['entity_id', 'uuid']:
+                if isinstance(field_value, VersionedModel):
+                    field_value = str(field_value.entity_id).replace('-', '')
+                elif isinstance(field_value, dict):
+                    field_value = str(field_value.get('entity_id')).replace('-', '')
+                elif isinstance(field_value, str):
+                    field_value = field_value.replace('-', '')
+            if isinstance(field_value, UUID):
+                field_value = str(field_value).replace('-', '')
+            if isinstance(field_value, datetime):
+                field_value = field_value.strftime('%Y-%m-%d %H:%M:%S')
+
+            data[field.name] = field_value
+        return data
+
+    def get_move_entity_to_audit_table_query(self, table, entity_id):
+        instance = self.get_one(table, "", {'entity_id': entity_id})
+        if instance:
+            self.db[f"{table}_audit"].insert_one(instance)
+    
+    def get_save_query(self, table, data):
+        self.db[table].find_one_and_update(
+            {'entity_id': data.get("entity_id")},
+            {'$set': data},
+        )
+    
+    def save(self, instance: VersionedModel, send_message: bool = False):
+        """Save func"""
+        data = self._process_data_before_save(instance)
+        with self.adapter:
+            self.adapter.run_transaction([lambda: self.get_move_entity_to_audit_table_query(self.table_name, data.get("entity_id")), lambda: self.get_save_query(self.table_name, data)])
+            if send_message:
+                # This assumes that the instance is now in post-saved state with all the new DB updates
+                message = json.dumps(instance.as_dict(convert_datetime_to_iso_string=True))
+                self.message_adapter.send_message(self.queue_name, message)
+        return instance
+
+    def _insert(self, data: Dict, collection_name: str):
+        self.db[collection_name].insert_one(data)
+
+    def _outdate(self, data: Dict, collection_name: str):
+        collection = self.db[collection_name]
+        return collection.find_one_and_update(
+            {'entity_id': data.get("entity_id"), 'latest': True, 'active': True},
+            {'$set': {'latest': False}},
```

Comment:
Based on [this comment in slack](https://ecorrouge.slack.com/archives/C05SKRK93M5/p1715703410212819), `latest` is not needed, the `_audit` collections should be used instead.

here is Jay's quote from the linked comment:
> One thing I forgot to mention, and I think that this is the place, is that the SurrealDB data adapter forced us to move away from using latest and into using _audit tables.  We couldn't support the graph relationships in SurrealDB without having the entity_id be the unique key.  Because of that, the MySQL support in rococo uses _audit tables as well and I think we want that to be the standard from here on.  That would mean that the MongoDB support would use _audit collections.